mp/src/game/client/clientmode_shared.cpp

```diff
@@ -1203,6 +1207,16 @@ void ClientModeShared::FireGameEvent( IGameEvent *event )
 		{
 			// that's me
 			pPlayer->TeamChange( team );
+#ifdef NEO
+			// Switch off Xray when switching teams
+			for (int i = 0; i < MAX_PLAYERS; i++)
```

Comment:
Could we switch the loop condition here to `i < gpGlobals->maxClients` to avoid looping too much for `sv_maxplayers` < 32